modules/apps/product-navigation/product-navigation-taglib/src/main/resources/META-INF/resources/personal_menu/js/PersonalMenu.es.js

```diff
@@ -20,33 +20,30 @@ import {fetch} from 'frontend-js-web';
 import PropTypes from 'prop-types';
 import React, {useRef, useState} from 'react';
 
-function mapItemsOnClick(items) {
-	return items.map((item) => {
-		if (item.items) {
-			item.items = [...mapItemsOnClick(item.items)];
-		}
+function mapItemOnClick(item) {
+	if (item.items) {
+		item.items = item.items.map(mapItemOnClick);
+	}
 
-		const {jsOnClickConfig, ...otherKeys} = item;
+	const jsOnClickConfig = item.jsOnClickConfig;
 
-		return {
-			onClick: jsOnClickConfig
-				? () => {
-						const {selectEventName, title, url} = jsOnClickConfig;
+	if (jsOnClickConfig) {
+		item.onClick = () => {
+			const {selectEventName, title, url} = jsOnClickConfig;
 
-						Liferay.Util.openSelectionModal({
-							id: selectEventName,
-							onSelect(selectedItem) {
-								Liferay.Util.navigate(selectedItem.url);
-							},
-							selectEventName,
-							title,
-							url,
-						});
-				  }
-				: undefined,
-			...otherKeys,
+			Liferay.Util.openSelectionModal({
+				id: selectEventName,
+				onSelect(selectedItem) {
+					Liferay.Util.navigate(selectedItem.url);
+				},
+				selectEventName,
+				title,
+				url,
+			});
 		};
-	});
+	}
+
+	return {...item};
```

Comment:
So this will actually throw some errors because you are including `jsOnClickConfig` into the item, which will get spread into the jsx element and will get added as a DOM attribute. So we need to explicitly exclude that key.